pkg/agent/agent.go

```diff
@@ -280,14 +280,21 @@ func (k *klocksmith) process(ctx context.Context) error {
 
 	klog.Infof("Deleting/Evicting %d pods", len(pods.Pods()))
 
-	if err := drainer.DeleteOrEvictPods(pods.Pods()); err != nil {
-		// Not ideal but its untyped and the only way this condition is propagated,
-		// When we're ready to hand over the reigns for timeouts we can drop this.
-		if strings.Contains(err.Error(), "global timeout reached") {
-			// Continue anyways, the reboot should terminate it.
-			klog.Warningf("Timed out draining node, continuing...")
-		} else {
-			return fmt.Errorf("deleting/evicting pods: %w", err)
+	drainErr := make(chan error)
```

Comment:
Let's have a buffer here so we don't block the goroutine from exiting if we never have a chance to read from the channel.
```suggestion
	drainErr := make(chan error, 1)
```